x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/TransportMountSearchableSnapshotAction.java

```diff
@@ -237,33 +237,32 @@ protected void masterOperation(
-                        // Restore the single index specified
-                        .indices(indexName)
-                        // Always rename it to the desired mounted index name
-                        .renamePattern(".+")
-                        .renameReplacement(mountedIndexName)
-                        // Pass through index settings, adding the index-level settings required to use searchable snapshots
-                        .indexSettings(indexSettings)
-                        // Pass through ignored index settings
-                        .ignoreIndexSettings(ignoreIndexSettings.toArray(new String[0]))
-                        // Don't include global state
-                        .includeGlobalState(false)
-                        // Don't include aliases
-                        .includeAliases(false)
-                        // Pass through the wait-for-completion flag
-                        .waitForCompletion(request.waitForCompletion())
-                        // Pass through the master-node timeout
-                        .masterNodeTimeout(request.masterNodeTimeout())
-                        // Fail the restore if the snapshot found above is swapped out from under us before the restore happens
-                        .snapshotUuid(snapshotId.getUUID())
-                        // Log snapshot restore at the DEBUG log level
-                        .quiet(true),
-                    delegate
-                );
+            RestoreSnapshotRequest restoreSnapshotRequest = new RestoreSnapshotRequest(repoName, snapName)
+                // Restore the single index specified
+                .indices(indexName)
+                // Always rename it to the desired mounted index name
+                .renamePattern(".+")
+                .renameReplacement(mountedIndexName)
+                // Pass through index settings, adding the index-level settings required to use searchable snapshots
+                .indexSettings(indexSettings)
+                // Pass through ignored index settings
+                .ignoreIndexSettings(ignoreIndexSettings.toArray(new String[0]))
+                // Don't include global state
+                .includeGlobalState(false)
+                // Don't include aliases
+                .includeAliases(false)
+                // Pass through the wait-for-completion flag
+                .waitForCompletion(request.waitForCompletion())
+                // Pass through the master-node timeout
+                .masterNodeTimeout(request.masterNodeTimeout())
+                // Fail the restore if the snapshot found above is swapped out from under us before the restore happens
+                .snapshotUuid(snapshotId.getUUID())
+                // Log snapshot restore at the DEBUG log level
+                .quiet(true);
+            // Specify the mount task as the parent of the refresh task
+            restoreSnapshotRequest.setParentTask(clusterService.localNode().getId(), task.getId());
```

Comment:
`setParentTask` doesn't follow the fluent pattern the other setters do, so this had to be separate

Reply:
You could use a `org.elasticsearch.client.internal.ParentTaskAssigningClient` if you really wanted a one-liner here.

Reply:
I think it's probably overkill in this case, but thanks!